controller/Controller.php

```diff
@@ -119,6 +125,126 @@ protected function returnError($code, $status, $message)
         echo "$code $status : $message";
     }
 
+    protected function notModified(Modifiable $modifiable = null)
+    {
+        $notModified = false;
+        if ($modifiable !== null && $modifiable->isUseModifiedDate()) {
+            $modifiedDate = $this->getModifiedDate($modifiable);
+            $notModified = $this->sendNotModifiedHeader($modifiedDate);
+        }
+        return $notModified;
+    }
+
+    /**
+     * Return the modified date.
+     *
+     * @param Modifiable $modifiable
+     * @return DateTime|null
+     */
+    protected function getModifiedDate(Modifiable $modifiable = null)
+    {
+        $modified = null;
+        $modifiedDate = $modifiable->getModifiedDate();
+        $gitModifiedDate = $this->getGitModifiedDate();
+        $configModifiedDate = $this->getConfigModifiedDate();
+
+        // max with an empty list raises an error and returns bool
+        if ($modifiedDate || $gitModifiedDate || $configModifiedDate) {
+            $modified = max($modifiedDate, $gitModifiedDate, $configModifiedDate);
+        }
+        return $modified;
+    }
+
+    /**
+     * Get the concept modified date. Returns the concept modified date if available. Otherwise,
+     * reverts to the modified date of the default concept scheme if available. Lastly, if it could
+     * not get the modified date from the concept nor from the concept scheme, it returns null.
+     *
+     * @param Concept $concept concept used to retrieve modified date
+     * @return DateTime|null|string|boolean
+     */
+    protected function getConceptModifiedDate(Concept $concept = null)
+    {
+        $modifiedDate = $concept->getModifiedDate();
+        if (!$modifiedDate) {
+            $vocab = $concept->getVocab();
+            $conceptSchemeURI = $vocab->getDefaultConceptScheme();
+            if ($conceptSchemeURI) {
+                $conceptSchemeGraph = $vocab->getConceptScheme($conceptSchemeURI);
+                if (!$conceptSchemeGraph->isEmpty()) {
+                    $literal = $conceptSchemeGraph->getLiteral($conceptSchemeURI, "dc:modified");
+                    if ($literal) {
+                        $modifiedDate = $literal->getValue();
```

Comment:
Here we use only the Concept date, or the date of the Vocabulary's main concept scheme. We never use the Vocabulary's dc:modified value.